solr/solr-ref-guide/src/nested-documents.adoc

```diff
@@ -36,17 +36,20 @@ Nested documents may be indexed via either the XML or JSON data syntax, and is a
 
 === Schema Notes
 
- * The schema must include indexed fields field `\_root_`. The value of that field is populated automatically and is the same for all documents in the block, regardless of the inheritance depth.
- Fields `\_nest_path_`, `\_nest_parent_` can be configured to store the path of the document in the hierarchy, and the unique `id` of the parent in the previous level.
- These 2 fields will be used by NestedUpdateProcessor URP, which is implicitly configured under Solr 8, when `\_root_` field is defined.
+ * The schema must include indexed field `\_root_`. The value of that field is populated automatically and is the same for all documents in the block, regardless of the inheritance depth. The id of the top document in every nested hierarchy is populated in this field.
+ * `\_nest_path_` can be configured to store the path of the document in the hierarchy
+ * `\_nest_parent_` can be configured to store the `id` of the parent in the previous level
  * Nested documents are very much documents in their own right even if certain nested documents hold different information from the parent.
    Therefore:
  ** the schema must be able to represent the fields of any document
  ** it may be infeasible to use `required`
  ** even child documents need a unique `id`
 
 
-=== Legacy Schema Notes
+=== Rudimentary Root-only schemas
+ * These schemas do not contain any other nested related fields apart for `\_root_`. +
+   In this mode relationship types(field names) between parents and their children are not saved.
+   All children are indexed under the `\_childDocuments_` field.
  * The schema must include an indexed, non-stored field `\_root_`. The value of that field is populated automatically and is the same for all documents in the block, regardless of the inheritance depth.
```

Comment:
I think you could also say here that with a root only schema, you must provide IDs for child documents since Solr will not compute one for you.  With the path fields, Solr _will_ compute one for you.

Reply:
It seems like it is not the case on master:
**JsonLoader#isChildDoc:617**

```java
private boolean isChildDoc(SolrInputDocument extendedFieldValue) {  
    return extendedFieldValue.containsKey(req.getSchema().getUniqueKeyField().getName());  
}
```
Perhaps we should file a new ticket for this one?

Reply:
Oh yeah; thanks for the reminder!   +1 to do a new ticket.

Reply:
BTW remember that it's only the JSON syntax that has this limitation here; we *do* support ID generation for XML and SolrJ.  We should state this.